src/NodeParser/IndexedAccessTypeNodeParser.ts

```diff
@@ -18,7 +19,13 @@ export class IndexedAccessTypeNodeParser implements SubNodeParser {
         return node.kind === ts.SyntaxKind.IndexedAccessType;
     }
     public createType(node: ts.IndexedAccessTypeNode, context: Context): BaseType {
-        const indexType = this.childNodeParser.createType(node.indexType, context);
+        let indexType = this.childNodeParser.createType(node.indexType, context);
+
+        if (indexType instanceof UnionType && indexType.getTypes().length === 1) {
```

Comment:
I think it would make more sense to fix this in `TypeOperatorNodeParser` so it doesn't create a UnionType in the first place when there is only one type.